server/app/services/geo/esri/FakeEsriClient.java

```diff
@@ -0,0 +1,155 @@
+public final class FakeEsriClient extends EsriClient {
+  private String jsonResources = System.getProperty("user.dir") + "/test/resources/esri/";
+  private JsonNode addressCandidates;
+  private JsonNode noAddressCandidates;
+  private JsonNode serviceAreaFeatures;
+  private JsonNode serviceAreaNoFeatures;
+  private JsonNode serviceAreaNotInArea;
+  private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+  @Inject
+  public FakeEsriClient(
+      Clock clock, EsriServiceAreaValidationConfig esriServiceAreaValidationConfig) {
+    super(clock, esriServiceAreaValidationConfig);
+  }
+
+  /**
+   * Returns address suggestions based on the provided address json.
+   *
+   * <p>The address key in the provided address json is used to determine which suggestions to
+   * return. The other fields do not matter.
+   *
+   * <ul>
+   *   <li>"Legit Address" – this value will provide three suggestions for tesing service area
+   *       validation: "Address In Area", "Address With No Service Area Features" and "Address Not
+   *       In Area"
+   *   <li>"Bogus Address" – use this value for testing address that won't return any suggestions
+   *   <li>"Error Address" – use this value to test what would happen if the external Esri service
+   *       were to return a non 200 status. The end result is the same as "Bogus Address"
+   *   <li>any other address will have the same effect as Bogus and Error
+   * </ul>
+   */
+  @Override
+  CompletionStage<Optional<JsonNode>> fetchAddressSuggestions(ObjectNode addressJson) {
+    String address = addressJson.findPath(AddressField.STREET.getValue()).textValue();
+    Optional<JsonNode> maybeJson = Optional.empty();
+    File resource;
+    FileInputStream inputStream;
+
+    switch (address) {
+      case "Legit Address":
+        if (addressCandidates != null) {
+          maybeJson = Optional.of(addressCandidates);
+          break;
+        }
+        try {
+          resource = new File(jsonResources + "findAddressCandidates.json");
+          inputStream = new FileInputStream(resource);
+          addressCandidates = Json.parse(inputStream);
+          maybeJson = Optional.of(addressCandidates);
+        } catch (FileNotFoundException e) {
+          logger.error("fakeEsriClient fetchAddressSuggestions file not found: {}", e);
+        }
+        break;
+      case "Bogus Address":
+        if (noAddressCandidates != null) {
+          maybeJson = Optional.of(noAddressCandidates);
+          break;
+        }
+
+        try {
+          resource = new File(jsonResources + "findAddressCandidatesNoCandidates.json");
+          inputStream = new FileInputStream(resource);
+          noAddressCandidates = Json.parse(inputStream);
+          maybeJson = Optional.of(noAddressCandidates);
+        } catch (FileNotFoundException e) {
+          logger.error("fakeEsriClient fetchAddressSuggestions file not found: {}", e);
+        }
+        break;
+      case "Error Address":
+      default:
```

Comment:
**suggestion (blocking):** at least log an error here, consider throwing an exception as well to signal the error state to the caller